httpx/middleware.py

```diff
@@ -147,17 +147,33 @@ def redirect_headers(self, request: AsyncRequest, url: URL, method: str) -> Head
         if url.origin != request.url.origin:
             # Strip Authorization headers when responses are redirected away from
             # the origin.
-            del headers["Authorization"]
+            self._strip_headers(headers)
             headers["Host"] = url.authority
 
         if method != request.method and method == "GET":
-            # If we've switch to a 'GET' request, then strip any headers which
-            # are only relevant to the request body.
-            del headers["Content-Length"]
-            del headers["Transfer-Encoding"]
+            self._delete_request_body_headers(headers)
 
         return headers
 
+    def _strip_headers(self, headers: Headers) -> None:
```

Comment:
Let's revert the changes moving the `del` logic into their own methods until the logic is needed in multiple places :)